### PR TITLE
Blogpost about dependency injection and Mockito 

### PR DESCRIPTION
**Pull request by jtokar**

### CODE REVIEW

**Comment by mureinik** [comment]
Nice writeup, but that's only half a job. The second half would be to send a patch that actually does this and link it from the blog-post.

^_^

**Comment by jtokar** [comment]
The patch that does it does a whole lot of other things as well, so I'm not sure it'll be very helpful...

File: source/blog/2016-06-29-dependency-injection-and-mockito.html.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+```
+
+You end up with something like this:
+ 
+```
+    @Mock
+    private HostedEngineHelper hostedEngineHelper;
+
+    @InjectMocks
+    public StorageDomainCommandBase<StorageDomainParametersBase> cmd = spy(new TestStorageCommandBase(new StorageDomainParametersBase()));
```
Comment (rmohr) [inline source/blog/2016-06-29-dependency-injection-and-mockito.html.md]: I think you can even write this:

```java
@InjectMocks
@Spy
public StorageDomainCommandBase<StorageDomainParametersBase> cmd = new TestStorageCommandBase(new StorageDomainParametersBase());
```

Comment (rgolangh) [inline source/blog/2016-06-29-dependency-injection-and-mockito.html.md]: you can give this as an example as 
backend/manager/modules/bll/src/test/java/org/ovirt/engine/core/bll/storage/domain/ImportHostedEngineStorageDomainCommandTest.java:

File: source/blog/2016-06-29-dependency-injection-and-mockito.html.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+---
+title: Dependency Injection and Mockito
+author: Jenny Tokar jtokar@redhat.com
+tags: community, blog, infrastructure
+date: 2016-06-29 15:00:00 CET
+---
+
+#Dependency Injection and Mockito
+
+Recently I encountered a small issue in a static helper class that  was full of static helper methods. After I finished dealing with the issue at hand and
```
Comment (rgolangh) [inline source/blog/2016-06-29-dependency-injection-and-mockito.html.md]: "that was" single space

File: source/blog/2016-06-29-dependency-injection-and-mockito.html.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            return hostedEngineHelper.isHostedEngineStorageDomain(storageDomain);
+    }
+```
+
+Those kind of wrapper methods are common in code that uses mocking testing frameworks for unit testing. They certainly are very easy to mock. However, code 
+shouldn’t exist for the sake of tests and unnecessary methods that are dirtying it should be removed. 
+And here comes the nice part: Mockito supports injecting mock classes into the tested class in a very neat way. 
+You simply add the helper class you need with the “Mock” annotation and put the “InjectMocks” annotation on the class you are mocking for testing and 
+that’s it. 
+Mockito will inject the mocked instance to the mocked class so you won’t fall on null pointer exceptions and you will be able to mock the methods you are
```
Comment (rgolangh) [inline source/blog/2016-06-29-dependency-injection-and-mockito.html.md]: I suggest inserting a header here saying something like "# Using @InjectMocks " or something similar for tldr readers

File: source/blog/2016-06-29-dependency-injection-and-mockito.html.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    
+    public boolean isHostedEngine(StorageDomain storageDomain){
+            return hostedEngineHelper.isHostedEngineStorageDomain(storageDomain);
+    }
+```
+
+Those kind of wrapper methods are common in code that uses mocking testing frameworks for unit testing. They certainly are very easy to mock. However, code 
+shouldn’t exist for the sake of tests and unnecessary methods that are dirtying it should be removed. 
+And here comes the nice part: Mockito supports injecting mock classes into the tested class in a very neat way. 
+You simply add the helper class you need with the “Mock” annotation and put the “InjectMocks” annotation on the class you are mocking for testing and
```
Comment (rgolangh) [inline source/blog/2016-06-29-dependency-injection-and-mockito.html.md]: wrap interesting keywords with backtics `` , and possibly make them bold to more emphasize them

File: source/blog/2016-06-29-dependency-injection-and-mockito.html.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+shouldn’t exist for the sake of tests and unnecessary methods that are dirtying it should be removed. 
+And here comes the nice part: Mockito supports injecting mock classes into the tested class in a very neat way. 
+You simply add the helper class you need with the “Mock” annotation and put the “InjectMocks” annotation on the class you are mocking for testing and 
+that’s it. 
+Mockito will inject the mocked instance to the mocked class so you won’t fall on null pointer exceptions and you will be able to mock the methods you are 
+not testing directly.
+
+So instead of having something like this in your test class::
+
+```
```
Comment (rgolangh) [inline source/blog/2016-06-29-dependency-injection-and-mockito.html.md]: use ```java. I think it should highlight the code nicely

File: source/blog/2016-06-29-dependency-injection-and-mockito.html.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+---
+title: Dependency Injection and Mockito
+author: Jenny Tokar jtokar@redhat.com
+tags: community, blog, infrastructure
```
Comment (rgolangh) [inline source/blog/2016-06-29-dependency-injection-and-mockito.html.md]: add unittest, java tags. @garrett any tag usage guidance?

**Comment by bproffitt** [comment]
Looping back; Is this blog ready, or should there be an addition of the patch as murenik suggested?

**Comment by jtokar** [comment]
It's ready.

**Comment by bproffitt** [comment]
Okay, a couple of things:

Your branch is stale now, so if you could go in and get it up to date with
the master branch, that would be great.

Also, can I get your first and last name for the byline?

Thanks,
BKP

On Sun, Jul 10, 2016 at 4:33 AM, jtokar <notifications@github.com> wrote:

> It's ready.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/oVirt/ovirt-site/pull/341#issuecomment-231577413>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AB6AZbt86sGwxXhvkuj1w-9Xc9NebPSWks5qUK4_gaJpZM4JBK7D>
> .
>



-- 
Brian Proffitt
Principal Community Analyst
Open Source and Standards
@TheTechScribe
574.383.9BKP

**Comment by jtokar** [comment]
Branch was updated. 
Name: Jenny Tokar

**Pull request merged**